common/static/common/js/spec/components/paging_collection_spec.js

```diff
@@ -0,0 +1,238 @@
+                collection.setSortField('test_field', true);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.DESCENDING);
+                collection.setSortField('test_field', true);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.ASCENDING);
+                collection.setSortField('test_field', true);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.DESCENDING);
+                collection.setSortField('test_field_2', true);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.DESCENDING);
+            });
+
+            SpecHelpers.withData({
+                'queries with page, page_size, and sort_order parameters when zero indexed': [true, 2],
+                'queries with page, page_size, and sort_order parameters when one indexed': [false, 3],
+            }, function (isZeroIndexed, page) {
+                collection.isZeroIndexed = isZeroIndexed;
+                collection.perPage = 5;
+                collection.sortField = 'test_field';
+                collection.setPage(3);
+                assertQueryParams({'page': page.toString(), 'page_size': '5', 'sort_order': 'test_field'});
+            });
+
+            SpecHelpers.withConfiguration({
+                'using a zero indexed collection': [true],
+                'using a one indexed collection': [false]
+            }, function (isZeroIndexed) {
+                collection.isZeroIndexed = isZeroIndexed;
+                server.isZeroIndexed = isZeroIndexed;
+            }, function () {
+                describe('setPage', function() {
+                    it('triggers a reset event when the page changes successfully', function () {
+                        var resetTriggered = false;
+                        collection.on('reset', function () { resetTriggered = true; });
+                        collection.setPage(3);
+                        server.respond();
+                        expect(resetTriggered).toBe(true);
+                    });
+
+                    it('triggers an error event when the requested page is out of range', function () {
+                        var errorTriggered = false;
+                        collection.on('error', function () { errorTriggered = true; });
+                        collection.setPage(17);
+                        server.respond();
+                        expect(errorTriggered).toBe(true);
+                    });
+
+                    it('triggers an error event if the server responds with a 500', function () {
+                        var errorTriggered = false;
+                        collection.on('error', function () { errorTriggered = true; });
+                        collection.setPage(3);
+                        AjaxHelpers.respondWithError(requests, 500, {}, requests.length - 1);
+                        expect(errorTriggered).toBe(true);
```

Comment:
It would be good to add some tests around the state of the collection if the server responds with an error. Does the collection's page remain as it was? Is the collection's state still consistent?